### PR TITLE
Actions update

### PR DESCRIPTION
**Pull request by MagicLike**
As seen in the warnings of GitHub Actions after running the [latest ci build](https://github.com/openandroidinstaller-dev/openandroidinstaller/actions/runs/5667887930), it needs an update.
grafik

TODO:

- [x] Update checkout
- [x] Update setup-python
- [x] Update automatic-releases
- [ ] ~~Update Python version(?)~~

---

See my latest test build here: https://github.com/MagicLike/openandroidinstaller/actions/runs/5685471674
grafik

### CODE REVIEW

**Comment by MagicLike** [comment]
@tsterbak I noticed, that OAI currently depends on Python 3.9 and does not allow 3.11. (https://github.com/openandroidinstaller-dev/openandroidinstaller/blob/main/requirements.txt) Should it even be considered to update to 3.11?

**Comment by tsterbak** [comment]
> @tsterbak I noticed, that OAI currently depends on Python 3.9 and does not allow 3.11. (https://github.com/openandroidinstaller-dev/openandroidinstaller/blob/main/requirements.txt) Should it even be considered to update to 3.11?

Thanks for the upgrade! :blush: 
The better source for requirements is the pyproject.toml, since I use poetry to manage the project: https://github.com/openandroidinstaller-dev/openandroidinstaller/blob/main/pyproject.toml
I would like to upgrade directly to 3.11 in the future, but right now some libraries lack compatibility as far as I can tell. So I would stay on 3.9 for now, since its still active and we don't need any features only available in newer versions (afaik).

**Comment by MagicLike** [comment]
> I would like to upgrade directly to 3.11 in the future, but right now some libraries lack compatibility as far as I can tell. So I would stay on 3.9 for now, since its still active and we don't need any features only available in newer versions (afaik).

If so - this PR is ready to ship! :tada:

File: .github/workflows/build-release.yml [hunk 1]
```diff
@@ -86,7 +86,7 @@ jobs:
     - name: Display structure of downloaded files
       run: ls -R
     - name: Create release
-      uses: marvinpinto/action-automatic-releases@latest
```
Comment (tsterbak) [inline .github/workflows/build-release.yml:89]: What is the reason for this change? Does it do the same thing?

Comment (MagicLike) [reply .github/workflows/build-release.yml:89]: It is a fork of the project of marvinpinto with multiple patches, such as switching to node 16. (GitHUb Action complains about the use of node 12 as it will no longer be supported at some point)

Comment (MagicLike) [reply .github/workflows/build-release.yml:89]: The only "problem" I had was the error seen in my second screenshot - maybe I can also resolve it, but afaik it depends on how you release a new version.

Comment (MagicLike) [reply .github/workflows/build-release.yml:89]: But it should do essentially the same - docs can be found here: https://github.com/dciborow/action-github-releases#supported-parameters

**Comment by MagicLike** [comment]
Now I get a different error message :D
grafik

**Comment by MagicLike** [comment]
@tsterbak Maybe you can try it on your end - Some how GitHub Actions is unable to access the releases of my fork...

Run with debug log: https://github.com/MagicLike/openandroidinstaller/actions/runs/5707419025/job/15465878236#step:11:213

**Comment by tsterbak** [comment]
> @tsterbak Maybe you can try it on your end - Some how GitHub Actions is unable to access the releases of my fork...

Yeah, this makes sense. I will just merge it and if necessary fix it at the time of the next release :) but it looks good! Thanks a lot! :)
